lib/spack/spack/util/web.py

```diff
@@ -286,6 +286,13 @@ def find_versions_of_archive(archive_urls, list_url=None, list_depth=0):
     for aurl in archive_urls:
         list_urls.add(spack.url.find_list_url(aurl))
 
+    # Add '/' to the end of the URL. Some web servers require this.
```

Comment:
@adamjstewart do you know if web servers ever *complain* about the trailing `/`? If not, perhaps `list_urls` could be updated to *replace* all paths without a trailing `/`. If that's not the case, or you don't know and are time-constrained to find out, then I'm inclined to merge this as is.

Reply:
Unfortunately, the reverse is also true. https://bitbucket.org/eigen/eigen/downloads/?tab=tags without a slash works just fine, but https://bitbucket.org/eigen/eigen/downloads/?tab=tags/ with a slash doesn't.